### DISCUSSION TITLE
[Question]: How to enable vitepress support with LSP alone.

### DISCUSSION THREAD

**Discussion by RayGuo-ergou** [vuejs/language-tools#4655] [Q&A] [answered]
It seems in https://github.com/vuejs/language-tools/pull/4321, the only way to enable vitepress is through config vscode extension, e.g. option `additionalExtensions` is removed. ( I might be wrong )

So my question is currently is there a way to enable with LSP alone if I config vue LSP to attach to markdown files? Tried the steps in the PR description without vscode step as I am not using it, but does not work.

**Comment by johnsoncodehk** [comment, marked as answer]
Setting `vue.server.includeLanguages` ​​in VSCode has two purposes:

1. Add markdown to the languages ​​of TS plugin: https://github.com/vuejs/language-tools/blob/923f835edb0119c0e85d6af388788c5c08e03a02/extensions/vscode/src/nodeClientMain.ts#L154
2. Add markdown to `LanguageClientOptions.documentSelector` of LanguageClient: https://github.com/vuejs/language-tools/blob/923f835edb0119c0e85d6af388788c5c08e03a02/extensions/vscode/src/common.ts#L217-L222

If you only use LSP (without hybrid mode), step 1 is not required.

**Reply by RayGuo-ergou** [reply]
Thanks for the reply!

However, "with LSP alone" what I mean by that is enable vitepress support in other editors e.g. neovim. It seems for now only vscode is supported as there's no option I can pass to LSP for enabling vitepress support?

**Reply by johnsoncodehk** [reply]
> there's no option I can pass to LSP

Do you mean passing to `@vue/language-server` or `LanguageClientOptions` in VSCode?

VitePress supports no need to pass options to `@vue/language-server`, all settings occur in tsconfig, `LanguageClientOptions`, TS plugin option.

btw we have a maintainers group on discord, if you want to join please message me on discord, my discord ID is #johnsoncodehk.

**Reply by RayGuo-ergou** [reply]
Oh I see! the missing part was I did not add markdown file type to ts plugin and tsserver, I got it working now. Thanks!

![image](https://github.com/user-attachments/assets/e2852202-2236-4e2c-9b09-28fb52453b00)

 > btw we have a maintainers group on discord, if you want to join please message me on discord, my discord ID is #johnsoncodehk.

Thanks! I have sent you friend request on discord :)

**Comment by sharpchen** [comment]
@RayGuo-ergou Mind sharing your config? I am struggling with this; I did a little search on your dotfile repo but failed to find it. Does this require a tsconfig.json?(I don't have one)

With the following config, volar can attach to markdown but no intellisense provided

```lua
require('lspconfig').ts_ls.setup({
  init_options = {
    plugins = {
      {
        name = '@vue/typescript-plugin',
        location = vim.fs.joinpath(
          vim.fs.dirname(vim.fs.dirname(vim.fn.system('echo -n $(readlink -f $(which vue-language-server))'))),
          'lib/node_modules/@vue/language-server'
        ),
        languages = { 'vue', 'markdown' },
      },
    },
  },
  filetypes = { 'typescript', 'javascript', 'javascriptreact', 'typescriptreact', 'vue' },
})

require('lspconfig').volar.setup({
  filetypes = { 'markdown' },
})

```

**Reply by RayGuo-ergou** [reply]
Hi @sharpchen , yes you do need to config `tsconfig.json` to make it work. see https://github.com/vuejs/language-tools/pull/4321 usage section. Btw you have to add `markdown` to `ts_ls`'s `filetypes` and you need to add `vue` to `volar`'s `filetypes` as it's override not merge.

**Reply by sharpchen** [reply]
Thanks, just made it work!
